pkg/kubelet/pod_workers.go

```diff
@@ -734,10 +751,75 @@ func calculateEffectiveGracePeriod(status *podSyncStatus, pod *v1.Pod, options *
 	return gracePeriod, status.gracePeriod != 0 && status.gracePeriod != gracePeriod
 }
 
+// allowPodStart tries to start the pod and returns true if allowed, otherwise
+// it requeues the pod and returns false.
+func (p *podWorkers) allowPodStart(pod *v1.Pod) bool {
+	if !kubetypes.IsStaticPod(pod) {
+		// TODO: Do we want to allow non-static pods with the same full name?
+		// Note that it may disable the force deletion of pods.
+		return true
+	}
+	p.podLock.Lock()
+	defer p.podLock.Unlock()
+	if !p.isStaticPodAllowedToStart(pod) {
+		p.workQueue.Enqueue(pod.UID, wait.Jitter(p.backOffPeriod, workerBackOffPeriodJitterFactor))
+		if status := p.podSyncStatuses[pod.UID]; status != nil {
```

Comment:
Instead of checking for nil, always use `ok` (there should never be a nil status)

```
if status, ok := p.podSyncStatuses[pod.UID]; ok {
  status.working = false
}
```